e2e-test/tsps_gcp_e2e_test.py

```diff
@@ -147,45 +144,91 @@ def upload_file_with_azcopy(sas_url):
         
         # upload the file
         with open(file=local_file_path, mode="rb") as blob_file:
+            data = blob_file.read()
             logging.info("preparing to upload blob")
-            blob_client.upload_blob(blob_file)
-
-
-# download a file from a sas url using the azure storage library
-def download_with_azcopy(sas_url):
-    blob_client = BlobClient.from_blob_url(sas_url)
-
+            headers = {
+                'Content-Type': 'application/octet-stream'
+            }
+            requests.put(signed_url, headers=headers, data=data)
+
+# download a file from a signed url
+def download_with_signed_url(signed_url):
+    # extract file name from signed url; signed url looks like:
+    # https://storage.googleapis.com/fc-secure-6970c3a9-dc92-436d-af3d-917bcb4cf05a/test_signed_urls/helloworld.txt?x-goog-signature...
+    local_file_name = signed_url.split("?")[0].split("/")[-1]
     # use a temporary directory that will get cleaned up after this block
     with tempfile.TemporaryDirectory() as tmpdirname:
-        local_file_name = blob_client.blob_name.split('/')[-1] # get the file name without directories
         local_file_path = os.path.join(tmpdirname, local_file_name)
         
-        # download the file
+        # download the file and write to local file
         with open(file=local_file_path, mode="wb") as blob_file:
-            download_stream = blob_client.download_blob()
-            blob_file.write(download_stream.readall())
+            download_stream = requests.get(signed_url).content
+            blob_file.write(download_stream)
+
+## GROUP MANAGEMENT FUNCTIONS
+def create_and_populate_terra_group(orch_url, group_name, group_admins, group_members, token):
+    # first create the group
+    group_email = create_terra_group(orch_url, group_name, token)
+
+    # now add admins and members
+    for email_address in group_admins:
+        add_member_to_terra_group(orch_url, group_name, email_address, "admin", token)
+
+    for email_address in group_members:
+        add_member_to_terra_group(orch_url, group_name, email_address, "member", token)
+    
+    return group_email
+
+
+def create_terra_group(orch_url, group_name, token):
+    uri = f"{orch_url}/api/groups/{group_name}"
+    headers = {
+        "Authorization": f"Bearer {token}",
+        "accept": "application/json"
+    }
+    response = requests.post(uri, headers=headers)
+    
+    if response.status_code != 201:
+        raise Exception(response.text)
+    
+    return response.json()['groupEmail']
+
+
+def add_member_to_terra_group(orch_url, group_name, email_address, role, token):
+    formatted_email = email_address.replace("@", "%40")
```

Comment:
can probably use urlencode here in case there are other weird characters in the values